apps/test/unit/promiseChainTest.js

```diff
@@ -0,0 +1,225 @@
+      expect(sequence).to.deep.equal(['A', 'B', 'C', 'D', 'E']);
+    });
+    sequence.push('A');
+    clock.tick(0); // Does nothing because tick has not been scheduled!
+    sequence.push('B');
+  });
+
+  it(`Promise "then" calls seem to be queued in the order they are received`, done => {
+    Promise.resolve().then(() => {
+      sequence.push('B');
+    });
+    sequence.push('A');
+    Promise.resolve().then(() => {
+      sequence.push('C');
+      expect(sequence).to.deep.equal(['A', 'B', 'C']);
+      done();
+    });
+  });
+
+  it(`But in complex situations this gets very hard to reason about`, done => {
+    const promiseChainBeingTested = Promise.resolve()
+        .then(() => {
+          return new Promise(resolve => setTimeout(() => {
+            sequence.push('A');
+            resolve();
+          }, 1));
+        })
+        .then(() => {
+          return new Promise(resolve => setTimeout(() => {
+            sequence.push('B');
+            resolve();
+          }, 1));
+        })
+        .then(() => {
+          sequence.push('C');
+        });
+
+    sequence.push('_');
+
+    const testCode = Promise.resolve()
+        .then(() => {
+          sequence.push('a');
+          clock.tick(1);
+        })
+        .then(() => {
+          sequence.push('b');
+        })
+        .then(() => {
+          sequence.push('c');
+          clock.tick(1);
+        })
+        .then(() => {
+          sequence.push('d');
+        });
+
+    Promise.all([promiseChainBeingTested, testCode]).then(() => {
+      expect(sequence).to.deep.equal(['_', 'a', 'A', 'b', 'c', 'B', 'd', 'C']);
+      done();
+    }).catch(done);
+  });
+
+  it(`especially when one chain has a mix of immediate and async promises.`, done => {
+    const promiseChainBeingTested = Promise.resolve()
+        .then(() => {
+          sequence.push('A');
+        })
+        .then(() => {
+          return new Promise(resolve => setTimeout(() => {
+            sequence.push('B');
+            resolve();
+          }, 1));
+        })
+        .then(() => {
+          sequence.push('C');
+        })
+        .then(() => {
+          return new Promise(resolve => setTimeout(() => {
+            sequence.push('D');
+            resolve();
+          }, 1));
+        });
+
+    sequence.push('_');
+
+    const testCode = Promise.resolve()
+        .then(() => {
+          sequence.push('a');
+        })
+        .then(() => {
+          sequence.push('b');
+          clock.tick(1);
+        })
+        .then(() => {
+          sequence.push('c');
+        })
+        .then(() => {
+          sequence.push('d');
+        })
+        .then(() => {
+          sequence.push('e');
+          clock.tick(1);
+        });
+
+    Promise.all([promiseChainBeingTested, testCode]).then(() => {
+      expect(sequence).to.deep.equal(['_', 'A', 'a', 'b', 'B', 'c', 'C', 'd', 'e', 'D']);
+      done();
+    }).catch(done);
+  });
```

Comment:
Again I'm not sure what this test is communicating. If you want to show something more, you could show that nesting `Promise.resolve().then()` calls has the same behavior as chaining the `.then()` calls.